stream.go

```diff
@@ -346,7 +347,14 @@ func (cs *clientStream) newAttemptLocked(sh stats.Handler, trInfo *traceInfo) (r
 	if err := cs.ctx.Err(); err != nil {
 		return toRPCErr(err)
 	}
-	t, done, err := cs.cc.getTransport(cs.ctx, cs.callInfo.failFast, cs.callHdr.Method)
+
+	// Add extra metadata (metadata that will be added by transport) to context
+	// so the balancer can see them.
+	ctx := grpcutil.WithExtraMetadata(cs.ctx, metadata.Pairs(
+		"content-type", grpcutil.ContentType(cs.callHdr.ContentSubtype),
+		"user-agent", cs.cc.dopts.copts.UserAgent,
+	))
```

Comment:
Should we only do this with the xds scheme to save some unnecessary context/metadata operations?

Reply:
Done

Reply:
Done